### DISCUSSION TITLE
Is MaxPooling necessary in encoder ?

### DISCUSSION THREAD

**Discussion by lchauvin** [voxelmorph/voxelmorph#622] [Q&A] [answered]
Is the MaxPooling necessary here, in the encoder:

```
        # encoder forward pass
        x_history = [x]
        for level, convs in enumerate(self.encoder):
            for conv in convs:
                x = conv(x)
            x_history.append(x)
            x = self.pooling[level](x)
```

given that we have strided convolution, and residual connections ?

What is the benefit of the max pooling ?

Thank you

**Comment by adalca** [comment, marked as answer]
In general, there's tons of architectural decisions that don't have huge impacts, and we've found that the exact choice isn't all that important. A strong (many feature) unet in general just goes a long way.
